packages/typespec-ts/test/modularUnit/scenarios/models/deserialization/propertyType.md

```diff
@@ -0,0 +1,623 @@
+# only: Should generate deserializer for property types
+Will prompt all operations into top-level.
+
+## TypeSpec
+
+This is tsp definition.
+
+```tsp
+model SimpleModel {
+  propString: string;
+  propboolean: boolean;
+  propNumber: int32;
+  propStringOptional?: string;
+  propSimpleUnion: string | boolean | int32;
+  propSimpleUnionOptional?: string | boolean | int32;
+  propStringLiteral: "A";
+  propBooleanLiteral: false;
+  propNumberLiteral: 1;
+  propStringLiteralOptional?: "A";
+  propStringUnion: "A" | "B";
+  propStringUnionOptioanl: "A" | "B";
+  propStringUnionNullable: "A" | "B" | null;
+  propStringUnionAsExtensible: "A" | "B" | string;
+  propStringUnionAsExtensibleOptional?: "A" | "B" | string;
+  propStringUnionAsExtensibleNullable: "A" | "B" | string | null;
+  propStringUnionAsExtensibleOptionalAndNullable?: "A" | "B" | string | null;
+  propMixedTypeLiteral: "A" | false | 1;
+  propStringArray: string[];
+  propBooleanArray: boolean[];
+  propNumberArray: int32[];
+  propSimpleUnionArray: (string | boolean | int32)[];
+  propStringArrayOptional?: string[];
+  propSimpleUnionArrayOptional?: (string | boolean | int32)[];
+  propRecordOfString: Record<string>;
+  propRecordOfBoolean: Record<boolean>;
+  propRecordOfNumber: Record<int32>;
+  propRecordOfSimpleUnion: Record<string | boolean | int32>;
+  propRecordOfStringOptional?: Record<string>;
+  propRecordOfStringArray: Record<string[]>;
+  propArrayOfRecordOfString: Record<string>[];
+  propArrayOfRecordOfStringOptional?: Record<string>[];
+  propRecordOfUnionArray: Record<(string | boolean | int32)[]>;
+  propRecordOfUnionArrayOptional?: Record<(string | boolean | int32)[]>;
+  propArrayOfRecordOfUnion: Record<string | boolean | int32>[];
+  propArrayOfRecordOfUnionOptional?: Record<string | boolean | int32>[];
+  @encodedName("application/json", "prop_encoded")
+  propEncoded: string;
+}
+
+@route("/serialize")
+interface D {
+  op bar(): { @body body: SimpleModel };
+}
+```
+
+This is the tspconfig.yaml.
+
+```yaml
+experimentalExtensibleEnums: true
+```
+
+## Provide generated models and its serializer
+
+Generated Models.
+
+```ts models
+/** model interface SimpleModel */
+export interface SimpleModel {
+  propString: string;
+  propboolean: boolean;
+  propNumber: number;
+  propStringOptional?: string;
+  propSimpleUnion: string | boolean | number;
+  propSimpleUnionOptional?: string | boolean | number;
+  propStringLiteral: "A";
+  propBooleanLiteral: false;
+  propNumberLiteral: 1;
+  propStringLiteralOptional?: "A";
+  propStringUnion: "A" | "B";
+  propStringUnionOptioanl: "A" | "B";
+  propStringUnionNullable: ("A" | "B") | null;
+  propStringUnionAsExtensible: "A" | "B";
```

Comment:
this generation should be incorrect as we already enabled experimentalExtensibleEnums for this case.